examples/cyclic-data.rs

```diff
@@ -0,0 +1,184 @@
+use ethercat::{
+    DomainIdx as DomainIndex, Idx, Master, MasterAccess, Offset, PdoCfg, PdoEntryIdx,
+    PdoEntryIdx as PdoEntryIndex, PdoEntryInfo, PdoEntryPos, PdoIdx, SlaveAddr, SlaveId, SlavePos,
+    SmCfg, SubIdx,
+};
+use ethercat_esi::EtherCatInfo;
+use std::{
+    collections::HashMap,
+    env,
+    fs::File,
+    io::{self, prelude::*},
+    thread,
+    time::Duration,
+};
+
+type BitLen = u8;
+
+pub fn main() -> Result<(), io::Error> {
+    env_logger::init();
+    let args: Vec<_> = env::args().collect();
+    let file_name = match args.len() {
+        2 => &args[1],
+        _ => {
+            println!("usage: {} ESI-FILE", env!("CARGO_PKG_NAME"));
+            return Ok(());
+        }
+    };
+
+    let mut esi_file = File::open(file_name)?;
+    let mut esi_xml_string = String::new();
+    esi_file.read_to_string(&mut esi_xml_string)?;
+    let esi = EtherCatInfo::from_xml_str(&esi_xml_string)?;
+    let (mut master, domain_idx, offsets) = init_master(&esi, 0_u32)?;
+    let cycle_time = Duration::from_micros(50_000);
+    master.activate()?;
+
+    loop {
+        master.receive()?;
+        master.domain(domain_idx).process()?;
+        master.domain(domain_idx).queue()?;
+        master.send()?;
+        let m_state = master.state()?;
+        let d_state = master.domain(domain_idx).state();
+        println!("Master state: {:?}", m_state);
+        println!("Domain state: {:?}", d_state);
+        if m_state.link_up && m_state.al_states == 8 {
+            let raw_data = master.domain_data(domain_idx);
+            println!("{:?}", raw_data);
+        }
+        thread::sleep(cycle_time);
+    }
+}
+
+pub fn init_master(
```

Comment:
This could (with some polishing) basically go into the lib, right?

Reply:
Indeed. But first I'd like to release a 0.2.0 version (see #12) and then we can add convenience functions step by step.